packages/core/strapi/lib/services/entity-service/components.js

```diff
@@ -43,9 +43,12 @@ const createComponents = async (uid, data) => {
           throw new Error('Expected an array to create repeatable component');
         }
 
-        const components = await Promise.all(
-          componentValue.map((value) => createComponent(componentUID, value))
-        );
+        // No Promise.all to avoid deadlocks
+        const components = [];
+        for (const value of componentValue) {
+          const compo = await createComponent(componentUID, value);
+          components.push(compo);
+        }
```

Comment:
I remember you and Alex talking about this PR https://github.com/strapi/strapi/pull/14389 . I understand that would prevent deadlocks like this one.

Does that PR needs more testing, or something we can help with?

If the idea is to use that in the future, could we add a TODO to use the Promise all again when we have the transaction api ready?

Reply:
Yes the other PR is not ready yet. I don't know exactly what it needs, it's Alex's work for the moment :)